pkg/manifests/tls.go

```diff
@@ -54,6 +54,39 @@ func (f *Factory) GRPCSecret() (*v1.Secret, error) {
 	return s, nil
 }
 
+func (f *Factory) MetricsClientCerts() (*v1.Secret, error) {
+	s, err := f.NewSecret(f.assets.MustNewAssetReader(ClusterMonitoringMetricsClientCertsSecret))
+	if err != nil {
+		return nil, err
+	}
+
+	s.Namespace = f.namespace
+	s.Data = make(map[string][]byte)
+	s.Annotations = make(map[string]string)
+
+	return s, nil
+}
+
+func (f *Factory) MetricsClientCASecret(apiAuthConfigmap *v1.ConfigMap) (*v1.Secret, error) {
```

Comment:
(nit) I would expect the function to accept []byte rather than configmap. Any reason why we pass a configmap here?

```suggestion
func (f *Factory) MetricsClientCASecret(clientCA []byte) (*v1.Secret, error) {
```

Reply:
this function more or less synchronizes the configmap from one namespace to another (the type is kept to be a configmap in my PR #1282).

I would keep the type of the argument in case we wanted to synchronize more fields in the future, given that it's a pointer, the size of the structure being passed should not differ that much.